File: objc/objcfind/Sources/objcfind/FindConfig.m
#import "FindConfig.h"

@implementation FindConfig

- (instancetype) init {
    self = [super init];
    if (self) {
        NSString *homePath = [[[NSProcessInfo processInfo] environment] objectForKey:@"HOME"];
        NSString *defaultXFindConfigDir = [NSString pathWithComponents:@[homePath, @".config", @"xfind"]];
        NSString *xfindConfigDir = [[[NSProcessInfo processInfo] environment] objectForKey:@"XFIND_CONFIG_DIR"];
        if (xfindConfigDir == nil) {
            xfindConfigDir = defaultXFindConfigDir;
        }
        NSString *defaultXFindPath = [NSString pathWithComponents:@[homePath, @"src", @"xfind"]];
        NSString *xfindPath = [[[NSProcessInfo processInfo] environment] objectForKey:@"XFIND_PATH"];
        if (xfindPath == nil) {
            xfindPath = defaultXFindPath;
        }
        self.xfindPath = xfindPath;
        self.fileTypesPath = [NSString pathWithComponents:@[xfindPath, @"shared", @"filetypes.json"]];
        self.findOptionsPath = [NSString pathWithComponents:@[xfindPath, @"shared", @"findoptions.json"]];
        self.defaultFindSettingsPath = [NSString pathWithComponents:@[xfindConfigDir, @"settings.json"]];
    }
    return self;
}

@end

File: swift/swiftfind/Sources/swiftfind/FindConfig.swift
//
//  FindConfig.swift
//  swiftfind
//
//  Created by Cary Clark on 5/12/15.
//  Copyright (c) 2015 Cary Clark. All rights reserved.
//

import Foundation

public struct FindConfig {
    public let xfindPath: String
    public let sharedPath: String
    public let fileTypesPath: String
    public let findOptionsPath: String
    public let defaultFindSettingsPath: String

    public init() {
        var defaultXFindConfigDir = "\(NSHomeDirectory())/.config/xfind"
        var xFindConfigDir: String
        if let xFindEnvConfigDir = ProcessInfo.processInfo.environment["XFIND_CONFIG_DIR"] {
            xFindConfigDir = xFindEnvConfigDir
        } else {
            xFindConfigDir = defaultXFindConfigDir
        }
        var defaultXFindPath = "\(NSHomeDirectory())/src/xfind"
        if let xFindEnvPath = ProcessInfo.processInfo.environment["XFIND_PATH"] {
            xfindPath = xFindEnvPath
        } else {
            xfindPath = defaultXFindPath
        }
        sharedPath = "\(xfindPath)/shared"
        fileTypesPath = "\(sharedPath)/filetypes.json"
        findOptionsPath = "\(sharedPath)/findoptions.json"
        defaultFindSettingsPath = "\(xFindConfigDir)/settings.json"
    }
}
